torchdp/per_sample_gradient_clip.py

```diff
@@ -171,7 +188,45 @@ def clip(self) -> List[float]:
         if self.stat is not None:
             stats.update(stats.StatType.CLIPPING, "ClippingStats", **self.stat)
             self.stat = {}
-        return threshs
+
+        if virtual:
+            # check if we're adding to an existing accumulator
+            if "clip_threshs" in self.accumulation_state:
+                # retain the largest clipping thresholds accross the entire batch
+                curr_thresh = self.accumulation_state["clip_threshs"]
+                self.accumulation_state["clip_threshs"] = [
+                    max(n1, n2) for (n1, n2) in zip(threshs, curr_thresh)
+                ]
```

Comment:
very neat! May be, I'd also leave the option of not changing the thresh and using the first batch thresh for the rest.  The reason being if someone uses a very small mini-batch norm, this could result in the a very large thresh to be used, for this diff may be you could simply add a TODO or warning here and leave it as is.

Reply:
agreed. I added a TODO.